### ISSUE TITLE
False positive: 'parameter must be non null' even though parameter was checked by Guavas Preconditions.checkNotNull()

### ISSUE THREAD

**Issue by ewirch** [spotbugs/spotbugs#580]
```    
int method(@Nullable String param) {
   Preconditions.checkNotNull(param);
   return param.length();
}
```

`checkNotNull()` will throw if `param` is null. So `param` is effectively non-null, when execution reaches the `return` statement.

Pull request containing test case follows....

**Comment by Sovietaced** [comment]
Same issue here. Had to roll our own `Preconditions.checkNotNull` with an annotation to suppress the FB warnings.

**Comment by KengoTODA** [comment]
See #581, you need to use `@NonNull` annotation instead of `@Nullable`.
